pylops_mpi/DistributedArray.py

```diff
@@ -1,12 +1,13 @@
-import numpy as np
-from typing import Optional, Union, Tuple, List
-from numbers import Integral
-from mpi4py import MPI
 from enum import Enum
+from numbers import Integral
+from typing import List, Optional, Tuple, Union
 
+import numpy as np
+from mpi4py import MPI
 from pylops.utils import DTypeLike, NDArray
 from pylops.utils._internal import _value_or_sized_to_tuple
-from pylops.utils.backend import get_module, get_array_module, get_module_name
+from pylops.utils.backend import get_array_module, get_module, get_module_name
+from pylops_mpi.utils.backend import nccl_split, nccl_allgather, nccl_allreduce, nccl_bcast, nccl_asarray
```

Comment:
This will currently break if I do not have `nccl` and `cupy` on my system.

```suggestion
try:
    from pylops_mpi.utils.backend import (
        nccl_split,
        nccl_allgather,
        nccl_allreduce,
        nccl_bcast,
        nccl_asarray,
    )
except ImportError:
    pass
```

Reply:
The reason it fails is likely due to the comments I put somewhere about having to change https://github.com/tharittk/pylops-mpi/blob/f1238cb0c26a90b672e2db38effb701dea1accee/pylops_mpi/utils/deps.py#L14 with a first condition checking if cupy is present.. I believe with that we should be good and can use the same code pattern to import those methods as for example here: https://github.com/PyLops/pylops/blob/44bae08e43c34516b71ac08e4e445e5835ddcc47/pylops/waveeqprocessing/twoway.py#L12

Reply:
Hmm, yup I forgot we already have `deps.nccl_import`, just updated the commit suggestion. It should now work fine after changing https://github.com/tharittk/pylops-mpi/blob/f1238cb0c26a90b672e2db38effb701dea1accee/pylops_mpi/utils/deps.py#L14 and applying the above suggestion change.